### PR TITLE
layers: Add descriptor buffer offset VU 09006

### PR DESCRIPTION
**Pull request by artem-lunarg**
Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/6050

### CODE REVIEW

**Review by juan-lunarg — APPROVED** [review]

**Review by spencer-lunarg — APPROVED** [review]

File: layers/core_checks/cc_descriptor.cpp [hunk 1]
```diff
@@ -3414,6 +3420,15 @@ bool CoreChecks::PreCallValidateCmdSetDescriptorBufferOffsetsEXT(VkCommandBuffer
         bool valid_buffer = false;
         bool valid_binding = false;
 
+        const auto set_layout = pipeline_layout->set_layouts[firstSet + i];
+        if ((set_layout->GetCreateFlags() & VK_DESCRIPTOR_SET_LAYOUT_CREATE_DESCRIPTOR_BUFFER_BIT_EXT) == 0) {
+            const LogObjectList objlist(cb_state->commandBuffer(), set_layout->GetDescriptorSetLayout(), pipeline_layout->layout());
+            skip |= LogError(objlist, "VUID-vkCmdSetDescriptorBufferOffsetsEXT-firstSet-09006",
+                             "vkCmdSetDescriptorBufferOffsetsEXT(): Descriptor set layout (%s) for set %" PRIu32
+                             " was created without VK_DESCRIPTOR_SET_LAYOUT_CREATE_DESCRIPTOR_BUFFER_BIT_EXT flag set",
+                             report_data->FormatHandle(set_layout->GetDescriptorSetLayout()).c_str(), i);
```
Comment (arno-lunarg) [inline layers/core_checks/cc_descriptor.cpp:3429]:
```suggestion
                             report_data->FormatHandle(set_layout->GetDescriptorSetLayout()).c_str(), firstSet + i);
```

File: tests/unit/descriptor_buffer.cpp [hunk 2]
```diff
@@ -744,6 +745,24 @@ TEST_F(NegativeDescriptorBuffer, BindingAndOffsets) {
                 command_buffer.end();
             }
         }
+
+        {
+            const VkDescriptorSetLayoutBinding binding = {0, VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1, VK_SHADER_STAGE_VERTEX_BIT,
+                                                          nullptr};
+            const auto set_layout_no_flag = VkDescriptorSetLayoutObj(m_device, {binding});
+            const auto pipeline_layout_2 = VkPipelineLayoutObj(m_device, {&set_layout_no_flag, &set_layout_no_flag});
```
Comment (arno-lunarg) [inline tests/unit/descriptor_buffer.cpp:753]:
```suggestion
            const VkDescriptorSetLayoutObj set_layout_no_flag(m_device, {binding});
            const VkPipelineLayoutObj pipeline_layout_2(m_device, {&set_layout_no_flag, &set_layout_no_flag});
```
But maybe there is a reason to write it the current way?

File: tests/unit/descriptor_buffer_positive.cpp [hunk 3]
```diff
@@ -55,4 +55,34 @@ TEST_F(PositiveDescriptorBuffer, BasicUsage) {
...
+    vk_testing::Buffer buffer(*m_device, buffer_ci, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT, VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT);
+
+    auto buffer_binding_info = LvlInitStruct<VkDescriptorBufferBindingInfoEXT>();
+    buffer_binding_info.address = buffer.address();
+    buffer_binding_info.usage = VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT;
+
+    const VkDescriptorSetLayoutBinding binding = {0, VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1, VK_SHADER_STAGE_VERTEX_BIT, nullptr};
+    const auto set_layout =
+        VkDescriptorSetLayoutObj(m_device, {binding}, VK_DESCRIPTOR_SET_LAYOUT_CREATE_DESCRIPTOR_BUFFER_BIT_EXT);
+    const auto pipeline_layout = VkPipelineLayoutObj(m_device, {&set_layout});
```
Comment (arno-lunarg) [inline tests/unit/descriptor_buffer_positive.cpp:79]: Same comment as above

**Review by arno-lunarg — COMMENTED** [review]
LGTM! 3 comments

File: layers/core_checks/cc_descriptor.cpp (hunk 1 above)
Comment (artem-lunarg) [reply layers/core_checks/cc_descriptor.cpp:3429]: thanks, good catch

File: tests/unit/descriptor_buffer.cpp (hunk 2 above)
Comment (artem-lunarg) [reply tests/unit/descriptor_buffer.cpp:753]: no specific reason, suggested version is cleaner

**Pull request merged**